src/connection-manager/wsHandling.ts

```diff
@@ -225,3 +235,69 @@ export async function handleWsMessageLedgerEnableAmendments(
     }),
   )
 }
+
+/**
+ * Backtracking a network to update amendment status in case of Websocket disconnection.
+ *
+ * @param network - The network being tracked.
+ * @param url - The Faucet URL of the network.
+ * @returns Void.
+ */
+async function backtrackNetworkAmendmentStatus(
+  network: string,
+  url: string,
+): Promise<void> {
+  try {
+    log.info(`Backtracking to update amendment status for ${network}...`)
+    const client = new Client(url)
+    await client.connect()
+    const ledgerResponse: LedgerResponse = await client.request({
+      command: 'ledger',
+      ledger_index: 'validated',
+    })
+    const currentLedger = ledgerResponse.result.ledger_index
+
+    // a flag + 1 ledger typically comes in every 10 to 15 minutes.
+    const fourFlagPlusOneLedgerBefore =
+      (Math.floor(currentLedger / 256) - 3) * 256 + 1
+
+    for (
+      let index = fourFlagPlusOneLedgerBefore;
+      index < currentLedger;
+      index += 256
+    ) {
+      const ledger: LedgerResponse = await client.request({
+        command: 'ledger',
+        transactions: true,
+        ledger_index: index,
+        expand: true,
+      })
+
+      await handleWsMessageLedgerEnableAmendments(
+        ledger as LedgerResponseCorrected,
```

Comment:
Can you avoid `as`?

Reply:
There's a missing field `hash` for the `BaseTransaction` in xrpl.js (which should be fixed), so I would need to cast to `LedgerResponseCorrected` which I created to capture the hash. Once this is fixed in our client library then there would be no need for custom field or casting.